src/service_control_client_impl.cc

```diff
@@ -267,32 +266,29 @@ void ServiceControlClientImpl::Quota(const AllocateQuotaRequest& quota_request,
 
   Status status = quota_aggregator_->Quota(quota_request, quota_response);
   if (status.code() == StatusCode::kNotFound) {
-    // Makes a copy of check_request so that on_done() callback can use
-    // it to call CacheResponse.
-    AllocateQuotaRequest* quota_request_copy =
-        new AllocateQuotaRequest(quota_request);
+    std::string quota_request_signature =
```

Comment:
Quota() call internally also calcualtes signature,  if we could pass it out, we could reduce a signature calculation But it may require more code changes. 

Cloud ESF may not switch away from this lib.  Not sure if it is worth the effort.

Reply:
I feel that switch is inevitable.  Maybe very soon.  Definitely not optimizing to reduce the sig-cal.   I am not even sure if we need this reducing request copying.

Reply:
I see, then at this point, we definitely don't want to spend that efforts for any further refactoring. Compared to the "right" refactoring, this change is a quick.

> I am not even sure if we need this reducing request copying.

I believe the main concern is whether this will break existing code. The risk is very small because `CacheResponse` is part of `aggregator_interface.h` which is only used internally to this package. Library users directly interact with the interfaces in `include/` folder, which has no change from this PR. 
Internally all the tests passed as well (cl/481384092).